test/hotspot/jtreg/compiler/vectorapi/VectorRearrangeTest.java

```diff
@@ -0,0 +1,222 @@
+    public void rearrange_int64() {
+        VectorShuffle<Integer> shuffle = VectorShuffle.fromArray(ispec64, indexes[0], 0);
+        for (int i = 0; i < LENGTH; i += ispec64.length()) {
+            IntVector.fromArray(ispec64, isrc, i)
+                     .rearrange(shuffle)
+                     .intoArray(idst, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.REARRANGE_VI, IRNode.VECTOR_SIZE_4, " >0 "})
+    public void rearrange_int128() {
+        VectorShuffle<Integer> shuffle = VectorShuffle.fromArray(ispec128, indexes[1], 0);
+        for (int i = 0; i < LENGTH; i += ispec128.length()) {
+            IntVector.fromArray(ispec128, isrc, i)
+                     .rearrange(shuffle)
+                     .intoArray(idst, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.REARRANGE_VL, IRNode.VECTOR_SIZE_2, " >0 "}, applyIfCPUFeature = {"asimd", "true"})
+    public void rearrange_long128() {
+        VectorShuffle<Long> shuffle = VectorShuffle.fromArray(lspec128, indexes[0], 0);
+        for (int i = 0; i < LENGTH; i += lspec128.length()) {
+            LongVector.fromArray(lspec128, lsrc, i)
+                      .rearrange(shuffle)
+                      .intoArray(ldst, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.REARRANGE_VF, IRNode.VECTOR_SIZE_2, " >0 "}, applyIfCPUFeature = {"asimd", "true"})
+    public void rearrange_float64() {
+        VectorShuffle<Float> shuffle = VectorShuffle.fromArray(fspec64, indexes[0], 0);
+        for (int i = 0; i < LENGTH; i += fspec64.length()) {
+            FloatVector.fromArray(fspec64, fsrc, i)
+                       .rearrange(shuffle)
+                       .intoArray(fdst, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.REARRANGE_VF, IRNode.VECTOR_SIZE_4, " >0 "})
+    public void rearrange_float128() {
+        VectorShuffle<Float> shuffle = VectorShuffle.fromArray(fspec128, indexes[1], 0);
+        for (int i = 0; i < LENGTH; i += fspec128.length()) {
+            FloatVector.fromArray(fspec128, fsrc, i)
+                       .rearrange(shuffle)
+                       .intoArray(fdst, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.REARRANGE_VD, IRNode.VECTOR_SIZE_2, " >0 "}, applyIfCPUFeature = {"asimd", "true"})
+    public void rearrange_double128() {
+        VectorShuffle<Double> shuffle = VectorShuffle.fromArray(dspec128, indexes[0], 0);
+        for (int i = 0; i < LENGTH; i += dspec128.length()) {
+            DoubleVector.fromArray(dspec128, dsrc, i)
+                        .rearrange(shuffle)
+                        .intoArray(ddst, i);
+        }
+    }
+
+    public static void main(String[] args) {
+        TestFramework testFramework = new TestFramework();
+        testFramework.setDefaultWarmup(10000)
+                     .addFlags("--add-modules=jdk.incubator.vector", "-XX:-TieredCompilation")
```

Comment:
```suggestion
                     .addFlags("--add-modules=jdk.incubator.vector")
```
I don't think that `TieredCompilation` should be necessary here. Or was it?